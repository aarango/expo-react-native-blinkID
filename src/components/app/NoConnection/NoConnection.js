import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { Colors } from '@auteco/theme';
import { styles } from './styles';
import { header as images } from '@auteco/assets';
import {
  View,
  Text,
  TouchableHighlight,
  Image,
  SafeAreaView,
  StatusBar,
} from 'react-native';

const NoConnection = ({ retryConnection, isInternetReachable }) => {
  const [buttonEnabled, setButtonEnabled] = useState(true);

  const setConnectionState = async () => {
    setButtonEnabled(false);
    if (await isInternetReachable()) {
      retryConnection();
    } else {
      setButtonEnabled(true);
    }
  };
  const disableButtonStyle = buttonEnabled
    ? {}
    : { backgroundColor: Colors.GRAY_LIGHT };
  return (
    <>
      <SafeAreaView style={styles.safeArea}>
        <StatusBar barStyle="dark-content" backgroundColor={Colors.WHITE} />
        <View style={styles.header}>
          <View style={styles.iconContainer}>
            <Image
              style={styles.image}
              source={images.logo}
              resizeMode="contain"
            ></Image>
          </View>
        </View>
        <View
          style={[styles.container, styles.verticallyCentered]}
          testID="NoConnection"
        >
          <Text style={[styles.textCentred, styles.title]}>Discúlpanos</Text>
          <Text style={[styles.textCentred, styles.mainText]}>
            Se presento un inconveniente tecnico.
          </Text>
          <Text style={[styles.textCentred, styles.mainText]}>
            Comprueba tu conexión a internet e intenta de nuevo.
          </Text>
          <TouchableHighlight
            onPress={buttonEnabled ? setConnectionState : () => {}}
            style={[styles.button, disableButtonStyle]}
            testID="button"
          >
            <Text style={[styles.textCentred, styles.buttonText]}>
              Reintentar
            </Text>
          </TouchableHighlight>
        </View>
      </SafeAreaView>
    </>
  );
};

NoConnection.propTypes = {
  retryConnection: PropTypes.func,
  isInternetReachable: PropTypes.func,
};

export { NoConnection };
