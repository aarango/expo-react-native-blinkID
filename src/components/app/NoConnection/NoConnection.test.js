import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import { NoConnection } from './NoConnection';

describe('<NoConnection />', () => {
  it('should match the snapshot', async () => {
    const retryConnection = jest.fn();

    const { toJSON, getByTestId } = render(
      <NoConnection retryConnection={retryConnection} />,
    );
    const button = getByTestId('button');
    fireEvent.press(button);
    expect(toJSON()).toMatchSnapshot();
  });
});
