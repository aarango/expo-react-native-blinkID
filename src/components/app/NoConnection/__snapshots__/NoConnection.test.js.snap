// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NoConnection /> should match the snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#000000",
        "flexDirection": "row",
        "height": 60,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "height": 50,
          "justifyContent": "center",
          "marginHorizontal": 20,
          "marginTop": 3,
          "width": 50,
        }
      }
    >
      <Image
        resizeMode="contain"
        source={1}
        style={
          Object {
            "height": 50,
            "width": 50,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "flex": 1,
        },
        Object {
          "justifyContent": "center",
        },
      ]
    }
    testID="NoConnection"
  >
    <Text
      style={
        Array [
          Object {
            "textAlign": "center",
          },
          Object {
            "color": "#000000",
            "fontSize": 25,
            "fontWeight": "bold",
            "lineHeight": 24,
            "marginBottom": 24,
          },
        ]
      }
    >
      Discúlpanos
    </Text>
    <Text
      style={
        Array [
          Object {
            "textAlign": "center",
          },
          Object {
            "alignSelf": "center",
            "fontSize": 18,
            "width": "80%",
          },
        ]
      }
    >
      Se presento un inconveniente tecnico.
    </Text>
    <Text
      style={
        Array [
          Object {
            "textAlign": "center",
          },
          Object {
            "alignSelf": "center",
            "fontSize": 18,
            "width": "80%",
          },
        ]
      }
    >
      Comprueba tu conexión a internet e intenta de nuevo.
    </Text>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignSelf": "center",
            "backgroundColor": "#0075DC",
            "borderRadius": 5,
            "height": 50,
            "justifyContent": "center",
            "marginTop": 35,
            "width": 200,
          },
          Object {
            "backgroundColor": "#A4A6AB",
          },
        ]
      }
      testID="button"
    >
      <Text
        style={
          Array [
            Object {
              "textAlign": "center",
            },
            Object {
              "color": "#FFFFFF",
              "fontSize": 18,
              "fontWeight": "bold",
            },
          ]
        }
      >
        Reintentar
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
